hw3/02-charts.js

```diff
@@ -1,52 +1,118 @@
 const backgroundColors = [
-  'rgba(54, 162, 235, 0.8)',
-  'rgba(255, 206, 86, 0.8)',
-  'rgba(255, 99, 132, 0.8)',
-  'rgba(75, 192, 192, 0.8)',
-  'rgba(153, 102, 255, 0.8)',
-  'rgba(255, 159, 64, 0.8)',
-  'rgba(199, 199, 199, 0.8)',
-  'rgba(83, 102, 255, 0.8)',
-  'rgba(40, 159, 64, 0.8)',
-  'rgba(210, 199, 199, 0.8)',
-  'rgba(78, 52, 199, 0.8)',
+  "rgba(54, 162, 235, 0.8)",
+  "rgba(255, 206, 86, 0.8)",
+  "rgba(255, 99, 132, 0.8)",
+  "rgba(75, 192, 192, 0.8)",
+  "rgba(153, 102, 255, 0.8)",
+  "rgba(255, 159, 64, 0.8)",
+  "rgba(199, 199, 199, 0.8)",
+  "rgba(83, 102, 255, 0.8)",
+  "rgba(40, 159, 64, 0.8)",
+  "rgba(210, 199, 199, 0.8)",
+  "rgba(78, 52, 199, 0.8)",
 ];
 
 const borderColors = [
-  'rgba(54, 162, 235, 1)',
-  'rgba(255, 206, 86, 1)',
-  'rgba(255, 99, 132, 1)',
-  'rgba(75, 192, 192, 1)',
-  'rgba(153, 102, 255, 1)',
-  'rgba(255, 159, 64, 1)',
-  'rgba(159, 159, 159, 1)',
-  'rgba(83, 102, 255, 1)',
-  'rgba(40, 159, 64, 1)',
-  'rgba(210, 199, 199, 1)',
-  'rgba(78, 52, 199, 1)',
+  "rgba(54, 162, 235, 1)",
+  "rgba(255, 206, 86, 1)",
+  "rgba(255, 99, 132, 1)",
+  "rgba(75, 192, 192, 1)",
+  "rgba(153, 102, 255, 1)",
+  "rgba(255, 159, 64, 1)",
+  "rgba(159, 159, 159, 1)",
+  "rgba(83, 102, 255, 1)",
+  "rgba(40, 159, 64, 1)",
+  "rgba(210, 199, 199, 1)",
+  "rgba(78, 52, 199, 1)",
 ];
 
 // url for the Thrones API
-const url = 'https://thronesapi.com/api/v2/Characters';
-
-const renderChart = () => {
-  const donutChart = document.querySelector('.donut-chart');
-
+const url = "https://thronesapi.com/api/v2/Characters";
+const donutChart = document.querySelector(".donut-chart");
+let housesObject = {};
+const errorMessage = document.createElement("h2");
+function cleanfamilyName(family) {
+  family = family.replace(/House/, "").trim();
+  if (family.includes("Lanister")) return "Lannister";
+  else if (family.includes("None")) return "Unknown";
+  else if (family.includes("Unkown")) return "Unknown";
+  else if (family.includes("Targaryan")) return "Targaryen";
+  else if (family.includes("Targaryn")) return "Targaryen";
+  else if (family === "") return "Unknown";
```

Comment:
This section can be improved a bit. For example, since lines 37, 38, and 41 all return "unknown", why not just test all the conditions in the same if statement using the logical or ( || ) operator.

Same thing for lines 39 and 40.